frontend/src/components/CreateEventModal.tsx

```diff
@@ -17,40 +17,53 @@ import {
     Select,
 } from '@chakra-ui/react';
 import DateTimePicker from 'react-datetime-picker';
-import { makeEvent } from '../controller/event';
-import { getMockedEventTypeList } from '../controller/eventType';
+import { UserContext } from '../contexts/UserContext';
+import { getEventTypeList } from '../controller/eventType';
+import { EventType } from '../../../src/typedefs/eventType';
+import { ErrorResponse } from '../../../src/typedefs/errorResponse';
+import { PostNewEventRequest } from '../../../src/typedefs/event';
 
-function generateEventTypeOptions(eventTypes: any[]) {
+interface CreateEventModalProps {
+    // eslint-disable-next-line no-unused-vars
+    createEventLocal: (newEvent: PostNewEventRequest) => void,
+}
+
+function generateEventTypeOptions(eventTypes: EventType[]) {
     const options: any[] = [];
     for (let i = 0; i < (eventTypes).length; i++) {
         options.push(<option key={i} value={eventTypes[i].eventTypeId}>{eventTypes[i].type}</option>);
     }
     return options;
 }
 
-async function handleClose(
-    eventName: string,
-    eventDescription: string,
-    startDateTime: Date,
-    endDateTime: Date,
-    eventTypeId: number,
-) {
-    await makeEvent(eventName, eventDescription, startDateTime, endDateTime, eventTypeId);
-}
-
-export default function CreateEventModal() {
+export default function CreateEventModal(props: CreateEventModalProps) {
+    const { state } = useContext(UserContext);
     const { isOpen, onOpen, onClose } = useDisclosure();
     const [startDateTime, setStartDateTime] = useState(new Date());
     const [endDateTime, setEndDateTime] = useState(new Date());
     const [eventName, setEventName] = useState('');
     const [description, setDescription] = useState('');
     const [eventTypeId, setEventTypeId] = useState(0);
-    const [eventTypes, setEventTypes] = useState([{}]);
+    const [eventTypes, setEventTypes] = useState<EventType[]>([]);
+
+    function getEventTypeListWasSuccessful(calendarEvent: EventType[] | ErrorResponse): calendarEvent is EventType[] {
+        if ((calendarEvent as EventType[]).length) {
+            return true;
+        }
+        // else, its an error
+        return false;
+    }
+
+    async function getEventTypeListLocal(token: string) {
+        const res = await getEventTypeList(token);
+        if (getEventTypeListWasSuccessful(res)) {
+            setEventTypes(res);
+        }
+    }
 
     useEffect(() => {
         async function getData() {
-            const types = await getMockedEventTypeList();
-            setEventTypes(types);
+            await getEventTypeListLocal(state.token);
```

Comment:
I'm not sure I understand, is this not in the effect?

Reply:
You're calling a function that is external to the effect, which is the problem - effect callbacks are optimized in part based on their dependency array, but functions that they call out too are not and can still potentially be referencing old values because of the way function declarations work in function components